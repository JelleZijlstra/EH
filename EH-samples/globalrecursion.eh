#!/usr/bin/ehi
# Can printvar handle this?
class A {
	public foo = $global
}
o = new A
printvar: $global
