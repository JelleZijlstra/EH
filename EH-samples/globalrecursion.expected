@object <AnonymousClass> [
A <public,non-static,non-constant>: @object <A> [
foo <public,non-static,non-constant>: (recursion)
]
CountClass <public,non-static,non-constant>: @object <CountClass> [
docount <public,non-static,non-constant>: @function <library method>: 
setcount <public,non-static,non-constant>: @function <library method>: 
]
File <public,non-static,non-constant>: @object <File> [
close <public,non-static,non-constant>: @function <library method>: 
getc <public,non-static,non-constant>: @function <library method>: 
gets <public,non-static,non-constant>: @function <library method>: 
open <public,non-static,non-constant>: @function <library method>: 
puts <public,non-static,non-constant>: @function <library method>: 
]
argc <public,non-static,non-constant>: @int 1
argv <public,non-static,non-constant>: @array [
0 => @string 'globalrecursion.eh'
]
class_is <public,non-static,non-constant>: @function <library>: 
eval <public,non-static,non-constant>: @function <library>: 
get_type <public,non-static,non-constant>: @function <library>: 
getinput <public,non-static,non-constant>: @function <library>: 
global <public,non-static,constant>: (recursion)
include <public,non-static,non-constant>: @function <library>: 
is_array <public,non-static,non-constant>: @function <library>: 
is_bool <public,non-static,non-constant>: @function <library>: 
is_float <public,non-static,non-constant>: @function <library>: 
is_int <public,non-static,non-constant>: @function <library>: 
is_null <public,non-static,non-constant>: @function <library>: 
is_object <public,non-static,non-constant>: @function <library>: 
is_range <public,non-static,non-constant>: @function <library>: 
is_string <public,non-static,non-constant>: @function <library>: 
log <public,non-static,non-constant>: @function <library>: 
o <public,non-static,non-constant>: @object <A> [
foo <public,non-static,non-constant>: (recursion)
]
pow <public,non-static,non-constant>: @function <library>: 
printvar <public,non-static,non-constant>: @function <library>: 
]
